polymorphism/task_3.py

```diff
@@ -0,0 +1,114 @@
+class Fraction:
+
+    def __init__(self,numerator,denominator):
+        if denominator==0:
+            raise ValueError("Denominator can't be equal zero.Please,try again.")
+        self.numerator=numerator
+        self.denominator=denominator
+
+
+    def __add__(self,other): #addition
+        if other==None or not isinstance(other,Fraction):
+            raise TypeError("Unexpected type")
+        
+        common_denominator=self.denominator*other.denominator
+        common_numerator=(self.denominator*other.numerator)+ (other.denominator*self.numerator)
+        return Fraction(common_numerator,common_denominator)
+
+    def __sub__(self,other):#subtraction
+        if other==None or not isinstance(other,Fraction):
+            raise TypeError("Unexpected type")
+        common_denominator=self.denominator*other.denominator
+        common_numerator=(self.denominator*other.numerator) - (other.denominator*self.numerator)
+        return Fraction(common_numerator,common_denominator)
+    
+    def __mul__(self,other):#multiplying
+        if other==None or not isinstance(other,Fraction):
+            raise TypeError("Unexpected type")
+        
+        common_denominator=self.denominator * other.denominator
+        common_numerator=self.numerator * other.numerator
+        return Fraction(common_numerator,common_denominator)
+    
+    def __truediv__(self,other):
+        if other==None or not isinstance(other,Fraction):
+            if other.denominator==0:
+                raise ZeroDivisionError("You can't divide by 0")
+            raise TypeError("Unexpected type")
+        
+        common_denominator=self.denominator*other.numerator
+        common_numerator=self.numerator*other.denominator
+        return Fraction(common_numerator,common_denominator)
+        
+
```

Comment:
It seems you are a big lover of empty lines in the code:)
Please, remove unnecessary empty lines